### PR TITLE
[IMP] l10n_ec: Update codes Tax 5% - May 2024

### PR DESCRIPTION
**Pull request by kenluzuriaga0**
Update tax codes:
- IVA 5% (411, Bienes)  ->  _IVA 5% (435, Bienes) (code_base: 435, code_applied: 445)_
- IVA 5% (510, Crédito IVA) -> _IVA 5% (550, Crédito IVA)  (code_base: 550, code_applied: 560)_

Create Report lines:
- _Ventas locales (excluye activos fijos) gravadas tarifa 5%_
- _Adquisiciones y pagos locales (excluye activos fijos) gravados con tarifa 5% (con derecho a crédito tributario)_

### CODE REVIEW

File: addons/l10n_ec/data/account_tax_report_data.xml [hunk 1]
```diff
@@ -52,6 +52,42 @@
...
+                                <field name="expression_ids">
+                                    <record id="tax_report_line_104_445_tag" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">tax_tags</field>
+                                        <field name="formula">445 (Reporte 104)</field>
+                                    </record>
+                                </field>
+                            </record>
+                        </field>
+                    </record>
```
[the remark this answers is not in the document]
Comment (kenluzuriaga0) [reply addons/l10n_ec/data/account_tax_report_data.xml]: Fixed and verified

**Comment by ren-odoo** [comment]
Hi @kenluzuriaga0 @stevTresCloud 

Thank you for the PR I did the FUN Testing and it's working correctly. Just can you update the PR description as the new tags are switched, it should be like this:
IVA 5% (411,, Bienes) -> IVA 5% (435, Bienes)
IVA 5% (510, Bienes) ->  IVA 5% (550, Bienes)

@jco-odoo this PR is ready for your review.

File: addons/l10n_ec/data/account_tax_report_data.xml [hunk 2]
```diff
@@ -87,6 +87,42 @@
                             </record>
                         </field>
                     </record>
+                    <record id="tax_report_line_parent_line_report_7" model="account.report.line">
+                        <field name="name">Ventas locales (excluye activos fijos) gravadas tarifa 5%</field>
```
Comment (aliyatastemirova) [inline addons/l10n_ec/data/account_tax_report_data.xml:91]: please note that starting from the next version in the forward port (saas-16.3) these report lines should be in English in the XML file and then we should add translations for these lines

**Review by aliyatastemirova — COMMENTED** [review]
hello, thank you for the pr! 
just wanted to leave a few comments, the review is up to @jco-odoo of course

File: addons/l10n_ec/migrations/3.9/post-migrate_update_taxes.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        for xml_id, l10n_ec_code_base, l10n_ec_code_applied in taxes_to_update:
+            cr.execute("UPDATE account_tax SET l10n_ec_code_base=%s, l10n_ec_code_applied=%s "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)",
+                       (l10n_ec_code_base, l10n_ec_code_applied, xml_id))
+
+
+def migrate(cr, version):
+    # For May Withholding Tax Code Updates
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    update_taxes_from_templates(cr, 'l10n_ec.l10n_ec_ifrs')
```
[the remark this answers is not in the document]
Comment (kenluzuriaga0) [reply addons/l10n_ec/migrations/3.9/post-migrate_update_taxes.py:45]: that method doesn't update the new l10n_ec fields. 
From 17.0 it do not need it. I did this in 17.0 https://github.com/odoo/odoo/pull/165672 without migration

File: addons/l10n_ec/data/account_tax_template_vat_data.xml [hunk 4]
```diff
@@ -7,40 +7,40 @@
         <!-- IVA 5% -->
         <record id="tax_vat_05_411_goods" model="account.tax.template">
```
Comment (stevTresCloud) [inline addons/l10n_ec/data/account_tax_template_vat_data.xml:8]: I don't know if the xml_id must be changed @kenluzuriaga0 @pepetreshere. The base code changed, so I suppose the xml_id changed too.

Comment (jco-odoo) [reply addons/l10n_ec/data/account_tax_template_vat_data.xml:8]: We can do in master

File: addons/l10n_ec/data/account_tax_report_data.xml [hunk 5]
```diff
@@ -87,6 +87,42 @@
...
+                        <field name="children_ids">
+                            <record id="tax_report_line_104_425" model="account.report.line">
+                                <field name="name">Valor bruto(425)</field>
+                                <field name="code">c425_104</field>
+                            <field name="expression_ids">
```
Comment (jco-odoo) [inline addons/l10n_ec/data/account_tax_report_data.xml:96]: indentation?

Comment (kenluzuriaga0) [reply addons/l10n_ec/data/account_tax_report_data.xml:96]: hmm... this identation is not shown  in two code editor

Comment (stevTresCloud) [reply addons/l10n_ec/data/account_tax_report_data.xml]: It's fixed @kenluzuriaga0. It's ready to merge @jco-odoo

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**